src/backend/distributed/shared_library_init.c

```diff
@@ -218,6 +218,13 @@ static const struct config_enum_entry explain_analyze_sort_method_options[] = {
 	{ NULL, 0, false }
 };
 
+static const struct config_enum_entry ddl_propagation_mode_options[] = {
+		{"parallelism", DDL_PROPAGATION_PARALLELISM, false},
+		{"optimistic",  DDL_PROPAGATION_OPTIMISTIC,  false},
+		{"always",      DDL_PROPAGATION_ALWAYS,      false},
```

Comment:
I find the naming a bit hard to digest

Thinking out loud:
citus.propagate_objects_in_transaction to 'immediate' / 'deferred' / 'automatic'

vaguely analogous to https://www.postgresql.org/docs/current/sql-set-constraints.html

Reply:
I think we only change `create_object`, because we cannot afford to do it for alter/drop etc.

So, maybe `citus.create_object_in_transaction`.

If the GUC/values also inform the user about the consequences, it'd be even more useful. 

So, other than us, no one can understand the meaning of `immediate` or `deferred` in this context.

Instead, 
1) on `immediate` we are fully transactional as long as the command succeeds
2) on `deferred` ROLLBACKs are broken
3) on `automatic` we are in between 1/2

so, _maybe_ : 
DDL_PROPAGATION_ALWAYS->`always_transactional`
DDL_PROPAGATION_PARALLELISM -> `sequentially_transactional` 
DDL_PROPAGATION_OPTIMISTIC -> `optimistically_transactional` 

These semantics are already guarded with `EnsureSequentialMode()` checks, so as long as the `create` does not throw error due to `EnsureSequentialMode`, we ensure the promise of `create_object_in_transaction` guc.

Reply:
Naming was something I was definitely not sure about, great discussion here!

I will follow the naming from Onder's suggestion as it is very explanatory.

Reply:
As a user, I don't really know how to understand optimistically_transactional or sequentially_transactional either.

Immediate at least communicates what happens: we propagate immediately.

Reply:
I thought some more about marco's proposition and it started to make more sense.
I have tweaked the guc name a bit. I didn't put transaction in the GUC name, instead I put it in their descriptions.